opts/options.go

```diff
@@ -244,3 +256,20 @@ func DisableValues() Option {
 		return nil
 	}
 }
+
+// QueryFilter sets a function that approves which peers may be dialed in a query
+func QueryFilter(filter func(h host.Host, ai peer.AddrInfo) bool) Option {
+	return func(o *Options) error {
+		o.QueryPeerFilter = filter
+		return nil
+	}
+}
+
+// RoutingTableFilter sets a function that approves which peers may be added to the routing table. The host should
+// already have at least one connection to the peer under consideration.
+func RoutingTableFilter(filter func(conns []core.Conn) bool) Option {
```

Comment:
Part of me wants to include a `host.Host` parameter in this function just in case we need it in the future. However, given that most users will probably just use our pre-built function breaking the API in the future may not be that big a deal.